File: rust/src/models/extra_component.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{price_entry::PriceEntry, product_ref::ProductRef};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct ExtraComponent {
    pub name: String,

    pub price_entry: PriceEntry,

    pub product_ref: ProductRef,
}

impl ExtraComponent {
    pub fn new(name: String, price_entry: PriceEntry, product_ref: ProductRef) -> Self {
        Self {
            name,
            price_entry,
            product_ref,
        }
    }
}

File: typescript/src/models/extraComponent.ts
// this file is @generated
import { type PriceEntry, PriceEntrySerializer } from "./priceEntry";
import { type ProductRef, ProductRefSerializer } from "./productRef";

export interface ExtraComponent {
  name: string;

  priceEntry: PriceEntry;

  productRef: ProductRef;
}

export const ExtraComponentSerializer = {
  _fromJsonObject(object: any): ExtraComponent {
    return {
      name: object["name"],
      priceEntry: PriceEntrySerializer._fromJsonObject(object["price_entry"]),
      productRef: ProductRefSerializer._fromJsonObject(object["product_ref"]),
    };
  },

  _toJsonObject(self: ExtraComponent): any {
    return {
      name: self.name,
      price_entry: PriceEntrySerializer._toJsonObject(self.priceEntry),
      product_ref: ProductRefSerializer._toJsonObject(self.productRef),
    };
  },
};
